### ISSUE TITLE
Require `$id`

### ISSUE THREAD

**Issue by gregsdennis** [json-schema-org/json-schema-linting#16]
All schemas should have an identifier.

**Comment by jdesrosiers** [comment]
I disagree with this one. I know the spec recommends that schemas include an identifier, but I've come to believe that's bad advice. I find it's better to use natural identifiers such as file paths or network locations. If a schema is truly in-memory-only, you can always provide an identifier when loading a schema rather than use `$id`.
